firewall.go

```diff
@@ -61,12 +98,17 @@ func New(opts *Options) (*Session, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	ret := &Session{
 		handle:     handle,
 		layerTypes: layerTypes{},
 	}
 
+	if opts.StartTransaction {
+		// Don't worry if we return an error here. We're not going to fail
+		// the creation of the session if we can't start a transaction.
+		ret.BeginTransaction(opts.TransactionFlags)
+	}
+
```

Comment:
Where's the commit transaction?

Reply:
We will abort an open transaction, see my comment below for details.